miio/gateway.py

```diff
@@ -529,45 +581,295 @@ def set_light(self, color_name, brightness):
+        except Exception as ex:
+            raise GatewayException(
+                "Got an exception while sending "
+                "command '%s' with arguments '%s'" % (command, str(arguments))
+            ) from ex
+
+    @command(click.argument("property"))
+    def get_subdevice_prop(self, property):
+        """Get the value of a property of the subdevice."""
+        try:
+            response = self._gw.send("get_device_prop", [self.sid, property])
+        except Exception as ex:
+            raise GatewayException(
+                "Got an exception while fetching property %s" % (property)
+            ) from ex
+
+        if not response:
+            raise GatewayException(
+                "Empty response while fetching property %s: %s" % (property, response)
+            )
+
+        return response
+
+    @command(click.argument("properties", nargs=-1))
+    def get_subdevice_prop_exp(self, properties):
+        """Get the value of a bunch of properties of the subdevice."""
+        try:
+            response = self._gw.send(
+                "get_device_prop_exp", [[self.sid] + list(properties)]
+            ).pop()
+        except Exception as ex:
+            raise GatewayException(
+                "Got an exception while fetching properties %s: %s" % (properties)
+            ) from ex
+
+        if len(list(properties)) != len(response):
+            raise GatewayException(
+                "unexpected result while fetching properties %s: %s"
+                % (properties, response)
+            )
+
+        for item in response:
+            if not item:
+                raise GatewayException(
+                    "One or more empty results while "
+                    "fetching properties %s: %s" % (properties, response)
+                )
+
+        return response
+
+    @command(click.argument("property"), click.argument("value"))
+    def set_subdevice_prop(self, property, value):
+        """Set a device property of the subdevice."""
+        try:
+            return self._gw.send("set_device_prop", {"sid": self.sid, property: value})
+        except Exception as ex:
+            raise GatewayException(
+                "Got an exception while setting propertie %s to value %s"
+                % (property, str(value))
+            ) from ex
+
+    @command()
+    def unpair(self):
+        """Unpair this device from the gateway."""
+        return self.subdevice_send("remove_device")
+
+    @command()
+    def get_battery(self):
+        """Update the battery level and return the new value."""
+        self._battery = self.subdevice_send("get_battery").pop()
+        return self._battery
+
+    @command()
+    def get_firmware_version(self) -> Optional[int]:
+        """Returns firmware version"""
+        return self.get_subdevice_prop("fw_ver").pop()
+
+
+class AqaraHT(SubDevice):
+    _temperature = None
+    _humidity = None
+    _pressure = None
```

Comment:
These need to be declared inside `__init__`, otherwise they are class attributes and are shared between objects (if you have multiple subdevices using the same class).

Reply:
If you have test devices for this, you could give dataclasses a try here. So simply convert this to:
```
@dataclass
class AqaraHT(SubDevice):
    temperature: int = None
    humidity: int = None
    pressure: int = None
```
remove the properties and set these values inside the update :-)

Reply:
I just commited an improved version, could you check if this is the way you imagened?